lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -95,7 +107,7 @@ export abstract class ClientApplication {
      * @returns {Promise.<AuthenticationResult | null>} token response or null. If the return value is null, then no auth redirect was detected.
      */
     async handleRedirectPromise(): Promise<AuthenticationResult | null> {
-        return this.handleRedirectResponse();
+        return this.isBrowserEnvironment ? this.handleRedirectResponse() : null;
```

Comment:
any chance this breaks client code that expects this to be a promise by returning null ?

Reply:
Since the function is declared as async this will resolve null. This essentially behaves the same as if there was no hash in the url